File: Codigos en C++/Ejercicio3_Ape06.cpp
#include <iostream> 
#include <vector> 
#include <string> 
#include <algorithm> // Librería para funciones como 'find'
 using namespace std; 
int main() { 
// 1. Declaración e inicialización rápida 
vector<string> productos = {"Laptop", "Mouse", "Teclado", "Monitor"}; 
string buscado; 
// 2. Captura de datos del usuario
cout << "Ingrese el producto a buscar: "; cin >> buscado; 
// 3. Búsqueda utilizando iteradores de la STL (Standard Template Library) 
auto it = find(productos.begin(), productos.end(), buscado); 
// 4. Validación del resultado 
if (it != productos.end()) { 
cout << "Resultado: El producto '" << buscado << "' SI se encuentra en stock." << endl; 
} else { 
    cout << "Resultado: Producto NO encontrado." << endl; 

}
return 0;
}

File: Codigos en Java/Ejercicio3_Ape06.java
import java.util.ArrayList; 
import java.util.Scanner;
public class Ejercicio3_Ape06 { 
public static void main(String[] args) { 
ArrayList<String> productos = new 
ArrayList<>(); 
Scanner scanner = new Scanner(System.in); // Objeto para lectura de teclado 
// 1. Población de datos 
productos.add("Laptop"); 
productos.add("Mouse"); 
productos.add("Teclado"); 
productos.add("Monitor"); 
// 2. Interacción con el usuario 
System.out.print("Ingrese el producto a buscar: "); String buscado = 
scanner.nextLine(); 
// 3. Verificación rápida usando el método nativo contains() 
if (productos.contains(buscado)) { 
System.out.println("Resultado: El producto '" + buscado + "' SÍ se encuentra en stock."); 
} else { 
System.out.println("Resultado: Producto NO encontrado."); 
} 
scanner.close(); // Liberación del recurso 
} 
}